counsel.el

```diff
@@ -3294,20 +3294,18 @@ attachment directory associated with the current buffer, all
 contained files are listed, so the return value could conceivably
 include attachments of other Org buffers."
   (require 'org-attach)
-  (let* ((ids (let (res)
-                (save-excursion
+  (let (files res dir)
+    (setq files (save-excursion
```

Comment:
It's redundant to have both `files` and `res` since they're the same thing:

```diff
diff --git a/counsel.el b/counsel.el
index 6d2be7c..3c39700 100644
--- a/counsel.el
+++ b/counsel.el
@@ -3294,24 +3294,23 @@ counsel-org-files
 contained files are listed, so the return value could conceivably
 include attachments of other Org buffers."
   (require 'org-attach)
-  (let (files res dir)
-    (setq files (save-excursion
-                  (goto-char (point-min))
-                  (while (re-search-forward "^:\\(ATTACH_DIR\\|ID\\):[\t ]+\\(.*\\)$" nil t)
-                    (setq dir (match-string-no-properties 2))
-                    (when (string= "ID" (match-string-no-properties 1))
-                      (setq dir (expand-file-name
-                                 (concat (substring dir 0 2) "/" (substring dir 2))
-                                 org-attach-directory)))
-                    (when (file-exists-p dir)
-                      (push dir res)))
-                  (nreverse res)))
+  (let (dirs)
+    (save-excursion
+      (goto-char (point-min))
+      (while (re-search-forward "^:\\(ATTACH_DIR\\|ID\\):[\t ]+\\(.*\\)$" nil t)
+        (let ((dir (match-string-no-properties 2)))
+          (when (string= "ID" (match-string-no-properties 1))
+            (setq dir (expand-file-name
+                       (concat (substring dir 0 2) "/" (substring dir 2))
+                       org-attach-directory)))
+          (when (file-exists-p dir)
+            (push dir dirs)))))
     (cl-mapcan
      (lambda (dir)
        (mapcar (lambda (file)
                  (file-relative-name (expand-file-name file dir)))
                (org-attach-file-list dir)))
-     files)))
+     (nreverse dirs))))
 
 ;;;###autoload
 (defun counsel-org-file ()
```

Reply:
Applied your patch, thanks.